### PR TITLE
podvm: support use different pause image when build podvm image

### PR DESCRIPTION
**Pull request by liudalibj**
This is a follow-up pr base on https://github.com/confidential-containers/cloud-api-adaptor/pull/1887

- update podvm-image dockerfiles so that we can use a different pause image for podvm
Put a docker auth file as `src/cloud-api-adaptor/podvm/auth.json` and run
```
export PAUSE_REPO=docker://my.auth.repo/pause-image
export PAUSE_VERSION=latest
export PAUSE_BIN=/usr/bin/pod
export AUTHFILE=auth.json
cd src/cloud-api-adaptor/
make podvm-builder podvm-binaries podvm-image
```

### CODE REVIEW

File: src/cloud-api-adaptor/Makefile [hunk 1]
```diff
@@ -186,6 +187,10 @@ podvm-binaries:
 	--build-arg PODVM_DISTRO=$(PODVM_DISTRO) \
 	--build-arg ARCH=$(ARCH) \
 	--build-arg ATTESTER=$(ATTESTER) \
+	--build-arg PAUSE_REPO=$(PAUSE_REPO) \
```
Comment (huoqifeng) [inline src/cloud-api-adaptor/Makefile:190]: Any default value for PAUSE_REPO and PAUSE_VERSION?

Comment (liudalibj) [reply src/cloud-api-adaptor/Makefile:190]: yeah, the default values are from version.yaml at here
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/Makefile.defaults#L44-L45
->  
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/versions.yaml#L49-L51

**Review by huoqifeng — APPROVED** [review]
LGTM
Thanks @liudalibj

**Review by mkulke — COMMENTED** [review]
does podvm-mkosi/Makefile need adjustments?

**Comment by liudalibj** [comment]
> does podvm-mkosi/Makefile need adjustments?

this change support fedora mkosi binaries build too.
Put a docker auth file as src/cloud-api-adaptor/podvm/auth.json and run
```
export PAUSE_REPO=docker://my.auth.repo/pause-image
export PAUSE_VERSION=latest
export PAUSE_BIN=/usr/bin/pod
export AUTHFILE=auth.json
cd src/cloud-api-adaptor/podvm-mkosi

make fedora-binaries-builder
make binaries
# we can check the pause_bundle at src/cloud-api-adaptor/podvm-mkosi/resources/binaries-tree/pause_bundle/
make image
```

**Review by mkulke — APPROVED** [review]

**Pull request merged**